xarray/backends/zarr.py

```diff
@@ -1167,7 +1167,7 @@ def _validate_and_autodetect_region(self, ds: Dataset) -> Dataset:
         region = self._write_region
 
         if region == "auto":
-            region = {dim: "auto" for dim in ds.dims}
+            region = dict.fromkeys(ds.dims, "auto")
```

Comment:
Why is this form better?

Reply:
Good question. I expect that if the rule exists, someone has measured run times and found this form more efficient. However, this has proven false at least once for [UP038](https://docs.astral.sh/ruff/rules/non-pep604-isinstance/) (see https://github.com/astral-sh/ruff/issues/7871).

From the [C420](https://docs.astral.sh/ruff/rules/unnecessary-dict-comprehension-for-iterable/) rule documentation:
> Prefer `dict.fromkeys(iterable)` over `{value: None for value in iterable}`, as `dict.fromkeys` is more readable and efficient.

Reply:
OK, I guess I'm familiar with the existing one, but support doing what is standard.

To the extent it's not yet a standard (that issue suggests it might not be yet), I'd vote with a low weight to push this off...

Reply:
I dropped `C420`.

Reply:
Great, looks good now